### ISSUE TITLE
About Params of EDSR-SR-STE

### ISSUE THREAD

**Issue by MingxiLi** [JungHunOh/SLS_CVPR2022#2]
Hi,

Thanks for your awesome work! I have a question about the number of parameters of EDSR-SR-STE.
For example, if we set N=8 and M=32, even if every conv can be replaced with a sparse_conv, the parameters of EDSR-SR-STE-N8M32 should be 8 / 32 * 1.52M = 380K. However, the reported params of EDSR-SR-STE-N8M32 is 346K, which is even less than x0.25 of the dense params. 
Do you adopt a modified version of EDSR? And will you release the architecture and pre-trained weights of Sparse EDSR models?

**Comment by JungHunOh** [comment]
Hi, thank you for your attention to our work!

We found that there were some mistakes when measuring the number of parameters of EDSR trained by SR-STE and one-shot pruning.
As you mentioned,  EDSR-SR-STE-N8M32 should be nearly 8 / 32 * 1.52M = 380K.
We will revise them in the arxiv version.
We also have checked the results from other model architectures and found that there are no more incorrect results.
Thank you for your careful comments.

**Comment by MingxiLi** [comment]
Hi, 

thanks for your reply! Is there a plan to release the pre-trained weights of EDSR-SLS? 
I tried to train EDSR-SLS with different sparsity target with the released code. However, the performance of EDSR-SLS-N4M32 and EDSR-SLS-N2M32 is not good. (while the performance of EDSR-SLS-N8M32 is the same as reported in the paper. )

**Comment by MingxiLi** [comment]
Hi, 

thanks for your reply! Is there a plan to release the pre-trained weights of EDSR-SLS? 
I tried to train EDSR-SLS with different sparsity target with the released code. However, the performance of EDSR-SLS-N4M32 and EDSR-SLS-N2M32 is not good. (while the performance of EDSR-SLS-N8M32 is the same as reported in the paper. )

**Comment by MingxiLi** [comment]
Hi, 

thanks for your reply! Is there a plan to release the pre-trained weights of EDSR-SLS? 
I tried to train EDSR-SLS with different sparsity target with the released code. However, the performance of EDSR-SLS-N4M32 and EDSR-SLS-N2M32 is not good. (while the performance of EDSR-SLS-N8M32 is the same as reported in the paper. )

**Comment by berrybuleming** [comment]
could you please explain your code:   binary_scores = binary_scores - scores.detach() + scores
没看懂啥意思

**Comment by JungHunOh** [comment]
Sorry for the delayed responses.
"binary_scores = binary_scores - scores.detach() + scores" is for the straight-through estimator (STE).
STE enables the differentiable binarization operation.

I'm sorry to say that we do not have any plans for releasing the pre-trained weights.
I think the results can be varied due to the different software or hardware.

Thanks :)
